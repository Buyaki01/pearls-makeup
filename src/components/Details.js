import Axios from 'axios';
import React, { useEffect, useState } from 'react';
import { useParams, NavLink } from 'react-router-dom';
import { IoMdArrowRoundBack } from '@react-icons/all-files/io/IoMdArrowRoundBack';

const Details = () => {
  const params = useParams();
  const { brand } = params;
  const [filteredMakeupArray, setfilteredMakeupArray] = useState([]);
  useEffect(async () => {
    const filterByBrand = await Axios.get(`http://makeup-api.herokuapp.com/api/v1/products.json?brand=${brand}`);
    const filteredMakeupData = await filterByBrand.data;
    for (let i = 0; i < filteredMakeupData.length; i += 1) {
      const title = filteredMakeupData[i].name;
      const { id } = filteredMakeupData[i];
      const priceSign = filteredMakeupData[i].price_sign;
      const { price } = filteredMakeupData[i];
      const { description } = filteredMakeupData[i];
      const image = filteredMakeupData[i].image_link;
      const { brand } = filteredMakeupData[i];
      const object = {
        title, id, priceSign, price, description, image, brand,
      };
      setfilteredMakeupArray((prevState) => [...prevState, object]);
    }
  }, []);
  return (
    <div className="detailsContainer">
      <NavLink to="/">
        <IoMdArrowRoundBack className="back-icon" />
      </NavLink>
      {filteredMakeupArray.map((makeup) => (
        <div key={makeup.id} className="detailsContents">
          <div className="images">
            <img alt="makeup" src={makeup.image} onError={(e) => { e.target.src = 'https://www.purpicks.com/wp-content/uploads/2018/02/Ombre-Amazonie-CC.png'; }} />
          </div>
          <div>
            <h3 className="makeupTitle">
              {makeup.title}
            </h3>
            <h5 className="makeupPrice">
              {makeup.priceSign}
              {makeup.price}
            </h5>
            <p className="description">{makeup.description}</p>
          </div>
        </div>
      ))}
    </div>
  );
};
export default Details;
