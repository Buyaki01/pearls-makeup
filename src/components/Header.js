import { FaPaintBrush } from 'react-icons/fa';

const Header = () => (
  <div className="header">
    <h1>
      <FaPaintBrush
        style={{
          color: 'white', fontSize: '30px', marginTop: '2px', marginRight: '2px',
        }}
      />
      Pearls Makeup
    </h1>
  </div>
);
export default Header;
