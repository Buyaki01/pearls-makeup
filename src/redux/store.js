import { createStore, combineReducers, applyMiddleware } from 'redux';
import logger from 'redux-logger';
import thunk from 'redux-thunk';
import makeupReducer from './makeupReducer';

const reducers = combineReducers({
  makeupReducer,
});

const store = createStore(
  reducers,
  applyMiddleware(logger, thunk),
);

export default store;
