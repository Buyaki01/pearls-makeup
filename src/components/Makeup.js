import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { getMakeupFromApi } from '../redux/makeupReducer';

const Makeup = () => {
  const makeupArray = useSelector((state) => state.makeupReducer);
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(getMakeupFromApi());
  }, []);
  return (
    <div>
      {makeupArray.map((makeup) => (
        <div key={makeup.id}>
          <img alt="makeup" src={makeup.image} />
          <div>
            <h3>
              {makeup.title}
            </h3>
            <h5>
              {makeup.priceSign}
              {makeup.price}
            </h5>
          </div>
        </div>
      ))}

    </div>
  );
};
export default Makeup;
