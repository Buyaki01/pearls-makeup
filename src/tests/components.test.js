import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import Header from '../components/Header';
import Makeup from '../components/Makeup';
import Details from '../components/Details';
import store from '../redux/store';

describe('Test UI of the webpage', () => {
  it('Displays the Header', () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Displays the Makeup page', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Makeup />
      </Provider>,
      { wrapper: MemoryRouter },
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('Displays the details page', () => {
    const tree = renderer.create(<Details />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
