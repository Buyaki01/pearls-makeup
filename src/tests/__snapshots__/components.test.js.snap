// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test UI of the webpage Displays the Header 1`] = `
<div
  className="header"
>
  <h1>
    Pearls Makeup
  </h1>
</div>
`;

exports[`Test UI of the webpage Displays the Makeup page 1`] = `
<DocumentFragment>
  <div
    class="header"
  >
    <h1>
      Pearls Makeup
    </h1>
  </div>
  <div
    class="makeUpContainer"
  />
</DocumentFragment>
`;

exports[`Test UI of the webpage Displays the details page 1`] = `
<div
  className="detailsContainer"
/>
`;
